src/plugins/intel_gpu/src/graph/impls/onednn/primitive_onednn_base.h

```diff
@@ -488,7 +488,18 @@ struct typed_primitive_onednn_impl : public typed_primitive_impl<PType> {
         }
 
         if (!instance.can_be_optimized()) {
-            _prim.execute(stream.get_onednn_stream(), _args[net_id]);
+            try {
+                _prim.execute(stream.get_onednn_stream(), _args[net_id]);
+            } catch (dnnl::error& err) {
+                /// WA: Force exit. Any opencl api call can be hang after CL_OUT_OF_RESOURCES.
+                if (err.status == dnnl_status_t::dnnl_out_of_memory) {
+                    std::cerr << "[GPU] WA: Force exit. CL_OUT_OF_RESOURCES is returned from call OpenCL API." << std::endl
+                              << "      It could occur hang issue." << std::endl
```

Comment:
Could you please mention here that the hang is the driver issue and recommend to try with the newer one? Like
`Due to the driver bug any subsequent OpenCL API call will cause application hang, so GPU plugin can't finish correctly. Please try to update the driver or reduce memory consumption (use smaller batch size, less streams, lower precision, etc) to avoid CL_OUT_OF_RESOURCES exception`